distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso

```diff
@@ -1175,6 +1177,109 @@ type Column
+
+       ? `Value_Type.Date`, `Value_Type.Date_Time`, `Value_Type.Time` format strings
+
+         A custom pattern string consists of one or more custom date and time
+         format specifiers. For example, "d MMM yyyy" will format "2011-12-03"
+         as "3 Dec 2011". See https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/time/format/DateTimeFormatter.html
+         for a complete format specification.
+
+         Note that the format string can specify decimal point and digit
+         separators, but these characters are interpreted in the context of the
+         Locale used. The format string specifies their location, but the Locale
+         has the final decision about which characters are used.
+
+       ? `Value_Type.Integer`, `Value_Type.Float` format strings
+
+         Numeric format strings are specified by the Java DecimalFormat class.
+         See https://docs.oracle.com/javase/8/docs/api/java/text/DecimalFormat.html
+         for a complete format specification.
+
+       ? `Value_Type.Boolean` format strings
+
+         Format strings for `Boolean` consist of two values that represent true
+         and false, separated by a `|`.
+
+       > Example
+         Format a `Column` of `Dates` in the format `"yyyyMMdd"`.
+
+             input = Column.from_vector "values" [Date.new 2020 12 21, Date.new 2023 4 25]
+             input.format "yyyyMMdd"
+             # ==> ["20201221", "20230425"]
+
+       > Example
+         Format `Column` of `Dates`, using format strings in a second column.
+
+             input = Column.from_vector "values" [Date.new 2020 12 21, Date.new 2023 4 25]
+             formats = Column.from_vector "formats" ["yyyyMMdd", "dd-MM-yyyy"]
+             input.format formats
+             # ==> ["20201221", "25-04-2023"]
+
+       > Example
+         Format a `Column` of `Integers` in the format `"#,##0.00"`.
+
+             input = Column.from_vector "values" [100000000, 2222, 3]
+             input.format "#,##0.00"
+             # ==> ["100,000,000.00", "2,222.00", "3.00"]
+
+       > Example
+         Format a `Column` of `Booleans` in the format `"t|f"`.
+
+            input = Column.from_vector "values" [True, False]
+            input.format "t|f"
+            # ==> ["t", "f"]
+    format : Text | Column -> Locale -> Column ! Illegal_Argument
+    format self format=Nothing locale=Locale.default =
+        create_formatter = case self.value_type of
+            Value_Type.Date -> make_value_formatter locale
+            Value_Type.Date_Time _ -> make_value_formatter locale
+            Value_Type.Time -> make_value_formatter locale
+            Value_Type.Boolean -> make_boolean_formatter
+            Value_Type.Integer _ -> make_value_formatter locale
+            Value_Type.Float _ -> make_value_formatter locale
+            bad_type ->
+                msg = "Cannot format a Column of type " + bad_type.to_text
+                Error.throw (Illegal_Argument.Error msg)
```

Comment:
might be nice to put this in `Column_Format`.

Reply:
Done.